lib/mongo/server/connection_pool/connection_pool_populator.rb

```diff
@@ -34,8 +34,15 @@ def pre_stop
 
       def do_work
         throw(:done) if @pool.closed?
-        unless @pool.populate
-          @pool.populate_semaphore.wait
+
+        begin
+          unless @pool.populate
+            @pool.populate_semaphore.wait
+          end
+        rescue Exception
```

Comment:
Let's change this to `rescue Error` to rescue all Mongo::Error derivatives, and `log_warn` the encountered exception prior to sleeping.